From 0c2dfc6776cd76d92eaeb95b75ebfa25ffa3105b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 13 Jul 2018 18:32:35 +0800
Subject: [PATCH 4261/5242] MLK-18605-8 drm/bridge: sec-dsim: add
 non-continuous clock support

commit  cfa8b438992faa1cab06885a403f72d405faac24 from
https://source.codeaurora.org/external/imx/linux-imx.git

Complete the non-continuous clock mode support.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit cd216df51f322434e45d12b6721637f1d8159fe4)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index a1575ee..e344dc8 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -521,8 +521,10 @@ static void sec_mipi_dsim_config_dpi(struct sec_mipi_dsim *dsim)
 
 	dsim_write(dsim, rgb_status, DSIM_RGB_STATUS);
 
-	if (dsim->mode_flags & MIPI_DSI_CLOCK_NON_CONTINUOUS)
+	if (dsim->mode_flags & MIPI_DSI_CLOCK_NON_CONTINUOUS) {
+		config |= CONFIG_NON_CONTINUOUS_CLOCK_LANE;
 		config |= CONFIG_CLKLANE_STOP_START;
+	}
 
 	if (dsim->mode_flags & MIPI_DSI_MODE_VSYNC_FLUSH)
 		config |= CONFIG_MFLUSH_VS;
-- 
1.7.9.5

